src/core/logical_layers/utils/logicalLayerFabric.ts

```diff
@@ -180,12 +168,16 @@ export function createLogicalLayerAtom(
             console.error('Download failed, source unavailable');
             return;
           }
-          downloadObject(
-            state.source.source,
-            `${
-              state.settings?.name || state.id || 'Disaster Ninja map layer'
-            }-${new Date().toISOString()}.json`,
-          );
+          if (state.source.source.type === 'geojson') {
+            downloadObject(
+              state.source.source.data,
+              `${
+                state.settings?.name || state.id || 'Disaster Ninja map layer'
+              }-${new Date().toISOString()}.json`,
+            );
+          } else {
+            console.error('Only geojson layers can be downloaded');
+          }
```

Comment:
I don't know should we tell it to user via toast?

Reply:
No, for users we hide download buttons, this check mostly for typescript